### PR TITLE
Fixes goroutine leak for failed connections

### PR DESCRIPTION
**Pull request by trevorlinton**
If a connection fails to establish a go routine is leaked from the Logger.writeLoop() running prior to returning logger,err on line 126/131.

You can test this by creating to go routines in a main function.   One that just waits for 10 seconds and prints out the amount of go routines, and the other that proceeds to try and connect to an bad host/port over and over with a dial/connect timeout of 1. 

Some psuedo code to demonstrate:

```
package main
import (
		"fmt"
		"github.com/papertrail/remote_syslog2/syslog"
		"time"
)
func main() {
	client : = ...
	host := ...
	t := time.NewTicker(time.Second * 10)
	for {
		syslog.Dial(client, "tcp", host, nil, time.Second*1, time.Second*2, 99990)
		fmt.Println("Go routines: ", runtime.NumGoroutine())
		<-t.C
        }
}
```

### CODE REVIEW

File: syslog/syslog.go [hunk 1]
```diff
@@ -109,6 +109,11 @@ func Dial(clientHostname, network, raddr string, rootCAs *x509.CertPool, connect
 	// dial once, just to make sure the network is working
 	conn, err := dial(network, raddr, rootCAs, connectTimeout)
 
+	// do not spawn a watch go routine if we cannot connect
+	if err != nil {
+		return nil, err
+	}
```
Comment (snorecone) [inline syslog/syslog.go]: I believe that it's intended to not return earlier here so the connection can be re-attempted within the `logger.writeLoop()` function.

**Comment by trevorlinton** [comment]
Ah, that makes more sense, so if we get an error back it could be an error that's temporary and could be ignored or it could be a bad hostname or something that will never resolve, but there may not be any way of knowing between either.

Should it just be a safe policy to run logger.Close() if we receive an err back?

In practice what was happening is we'd have bad hostnames end up in the syslog end point, and when it failed we didn't call logger.Close() as we though that the err being populated would insinuate the logger was nil like in golangs default net/dail.

**Comment by snorecone** [comment]
Yeah the code could definitely use some work in that respect, and you're right, there currently is not a way to tell if an error will be permanent or not. You could do as you describe and inspect the error and optionally close the logger.

**Comment by trevorlinton** [comment]
i'll chalk this up to user error on my part.  Thanks for the explanations!

**Pull request closed without merging**